api/source/graphql/mutations/ApplicationMutations.ts

```diff
@@ -0,0 +1,108 @@
+// (c) Tecnologico de Monterrey 2022, rights reserved.
+
+import { ApplicationType} from "../../types/ApplicationType";
+import { GraphQLBoolean, GraphQLError, GraphQLID, GraphQLNonNull, GraphQLString } from "graphql";
+import { v4 as uuid } from "uuid";
+import { APPLICATION_STATUS_TABLE_NAME, APPLICATION_TABLE_NAME, CITATION_TABLE_NAME, USER_TABLE_NAME } from "../../database/utils/database_constants";
+import { db } from "../../database/database";
+import { ApplicationStatusType } from "../../types/ApplicationStatusType";
+
+
+export default {
+
+  createApplication: {
+    type: ApplicationType,
+    args: {
+      user_id: {
+        type: GraphQLNonNull(GraphQLID),
+      },
+      title: {
+        type: GraphQLNonNull(GraphQLString)
+      },
+      deadline: {
+        type: GraphQLString,
+      },
+      end_time: {
+        type: GraphQLString,
+      },
+      emission_date: {
+        type: GraphQLNonNull(GraphQLString),
+      },
+      response_date: {
+        type: GraphQLNonNull(GraphQLString)
+      },
+      application_status_id: {
+        type: GraphQLNonNull(GraphQLID),
+      },
+      citation_id: {
+        type: GraphQLNonNull(GraphQLID),
+      }
+
+    },
+    resolve: async (_: any, { user_id,title,deadline, start_time,end_time, emission_date, response_date,
+    application_status_id,citation_id }: any) => {
+      const id = uuid();
+      await db(APPLICATION_TABLE_NAME)
+        .insert({
+          id,
+          title,
+          deadline,
+          start_time,
+          end_time,
+          emission_date, 
+          response_date, 
+          user_id,
+          application_status_id,
+          citation_id,
+        })
+        .catch((error: Error) => {
+          console.error(error);
+          throw new GraphQLError(error.name);
+        });
+      const newApplication = await db.select().from(APPLICATION_TABLE_NAME).where("id", id);
+      const myUser = await db.select().from(USER_TABLE_NAME).where({ id: user_id });
+      const myApplytatus = await db.select().from(APPLICATION_STATUS_TABLE_NAME).where({ id: application_status_id});
+      const myCitation = await db.select().from(CITATION_TABLE_NAME).where({ id: citation_id });
+
```

Comment:
Add validations, you are inserting all relationships `id`s first and then pulling each object. If `user_id`/`application_status_id`/`citation_id` does not exists, it will break the query, therefore a bunch of errors will show up.